Gems/Atom/RHI/Vulkan/Code/Source/RHI/StreamingImagePool.cpp

```diff
@@ -76,11 +76,18 @@ namespace AZ
                 return RHI::ResultCode::OutOfMemory;
             }
 
+            auto memoryView = m_memoryAllocator.Allocate(memoryRequirements.size, memoryRequirements.alignment);
```

Comment:
I know the [IAllocatorSchema::Allocator](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Memory/IAllocator.h#L36-L42) function returns a `void*`, so what is the type that m_memoryAllocator is returning?
It must valid class type, since it has an `IsValid` function on it, but it is not a pointer because it is used the dot operator.

Reply:
In this case `m_memoryAllocator` is an instance of the `MemoryTypeAllocator` template from here Gems\Atom\RHI\Vulkan\Code\Source\RHI\MemoryTypeAllocator.h. It is not derived from IAllocatorSchema::Allocator

`using MemoryAllocator = MemoryTypeAllocator<RHI::MemorySubAllocator<MemoryFreeListSubAllocatorTraits>, MemoryView>;`

The `Allocate` function returns an `AZ::Vulkan::MemoryView`.